compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.td

```diff
@@ -662,6 +662,22 @@ def AMDAIESinkIntoCore :
   let constructor = "mlir::iree_compiler::AMDAIE::createAMDAIESinkIntoCorePass()";
 }
 
+def AMDAIESplitLogicalObjFifos :
+  Pass<"iree-amdaie-split-logical-objectfifos", "ModuleOp"> {
+  let summary = "Pass to split L2 buffers to distribute on multiple shimTiles and memTiles.";
+  let description = [{
+    Splitting L2 input and output logical objectFifos and their user dma operations
+    by a factor of the number of AIE columns being used, so that the logical objectFifos
```

Comment:
No, but there's still a relationship with the number of columns. Suppose `rows: 4, cols: 2` and a different input being broadcasted to each row, then you likely still want to split it in `2` (number of columns), instead of 4.

Reply:
Exactly, that makes sense. Currently though it'd get split 4 ways afaict (I know the plan is to revisit this which is fine by me).

Reply:
> Currently though it'd get split 4 ways afaict

Yes, indeed, and that's a 'valid' splitting, however, it will likely fail because of not enough channels.